web/src/js/components/ContentView/ViewSelector.jsx

```diff
@@ -1,47 +1,82 @@
-import React, { PropTypes } from 'react'
+import React, { PropTypes, Component } from 'react'
 import classnames from 'classnames'
 import { connect } from 'react-redux'
 import * as ContentViews from './ContentViews'
-import { setContentView } from "../../ducks/ui/flow";
+import { setContentView, setContentViewSelectorOpen } from "../../ducks/ui/flow";
 
 
-function ViewButton({ name, setContentView, children, activeView }) {
+function ViewItem({ name, setContentView, children }) {
     return (
-        <button
-            onClick={() => setContentView(name)}
-            className={classnames('btn btn-default', { active: name === activeView })}>
-            {children}
-        </button>
+        <li>
+            <a href="#" onClick={() => setContentView(name)}>
+                {children}
+            </a>
+        </li>
     )
 }
-ViewButton = connect(state => ({
-    activeView: state.ui.flow.contentView
-}), {
-    setContentView
-})(ViewButton)
 
 
-ViewSelector.propTypes = {
-    message: PropTypes.object.isRequired,
-}
-export default function ViewSelector({ message }) {
+/*ViewSelector.propTypes = {
+    contentViews: PropTypes.array.isRequired,
+    activeView: PropTypes.string.isRequired,
+    isEdit: PropTypes.bool.isRequired,
+    isContentViewSelectorOpen: PropTypes.bool.isRequired,
+    setContentViewSelectorOpen: PropTypes.func.isRequired
+}*/
 
-    let autoView = ContentViews.ViewAuto.findView(message)
-    let autoViewName = (autoView.displayName || autoView.name)
-        .toLowerCase()
-        .replace('view', '')
-        .replace(/ContentLoader\((.+)\)/,"$1")
 
-    return (
-        <div className="view-selector btn-group btn-group-xs">
+class ViewSelector extends Component {
+      constructor(props, context) {
+        super(props, context)
+        this.close = this.close.bind(this)
+    }
+    close() {
+        this.props.setContentViewSelectorOpen(false)
+        document.removeEventListener('click', this.close)
+    }
 
-            <ViewButton name="ViewAuto">auto: {autoViewName}</ViewButton>
+    onDropdown(e){
+        e.preventDefault()
+        this.props.setContentViewSelectorOpen(!this.props.isContentViewSelectorOpen)
```

Comment:
Ugh. I appreciate the effort, but let's keep that in the plain react state - there's no way that we ever need this outside of this component.